.github/workflows/publish_pypi.yml

```
@@ -2,251 +2,38 @@ name: Build and publish package to PyPI
-          name: wheels_manylinux
-          path: ./wheelhouse/*.whl
-          if-no-files-found: error
-
-  build_macos_wheels:
-    name: Wheels (${{ matrix.os }})
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [macos-13, macos-14]
-      fail-fast: false
-    steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-      - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
-        with:
-          python-version: '3.11'
-
-      - name: Install cibuildwheel
-        run: python -m pip install cibuildwheel
-
-      - name: Build wheels on macOS
-        shell: bash
-        run: |
-            set -e -x
-
-            # cibuildwheel not recognising its environment variable, so set manually
-            export CIBUILDWHEEL="1"
-
-            python -m cibuildwheel --output-dir wheelhouse
-        env:
-          CIBW_ENVIRONMENT: >
-            PYBAMM_DISABLE_TELEMETRY="true"
-          # 10.13 for Intel (macos-13), 11.0 for Apple Silicon (macos-14 and macos-latest)
-          MACOSX_DEPLOYMENT_TARGET: ${{ matrix.os == 'macos-14' && '11.0' || '10.13' }}
-          CIBW_ARCHS_MACOS: auto
-          CIBW_BEFORE_BUILD: python -m pip install setuptools delocate
-          CIBW_REPAIR_WHEEL_COMMAND: |
-            if [[ $(uname -m) == "x86_64" ]]; then
-              delocate-listdeps {wheel} && delocate-wheel -v -w {dest_dir} {wheel}
-            elif [[ $(uname -m) == "arm64" ]]; then
-              # Use higher macOS target for now since casadi/libc++.1.0.dylib is still not fixed
-              delocate-listdeps {wheel} && delocate-wheel -v -w {dest_dir} {wheel} --require-target-macos-version 11.1
-              for file in {dest_dir}/*.whl; do mv "$file" "${file//macosx_11_1/macosx_11_0}"; done
-            fi
-          CIBW_TEST_EXTRAS: "all,dev,jax"
-          CIBW_TEST_COMMAND: |
-            set -e -x
-            python -m pytest -m cibw {project}/tests/unit
-
-      - name: Upload wheels for macOS (amd64, arm64)
-        uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
-        with:
-          name: wheels_${{ matrix.os }}
-          path: ./wheelhouse/*.whl
-          if-no-files-found: error
-
-  build_sdist:
-    name: Build SDist
+  Build:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Set up Python
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
         with:
           python-version: 3.12
 
-      - name: Build SDist
-        run: pipx run build --sdist
+      - name: Build wheel
+        run: |
+          pipx run build --outdir deploy
 
-      - name: Upload SDist
+      - name: Upload package
         uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
         with:
-          name: sdist
-          path: ./dist/*.tar.gz
+          name: wheels
```

Nitpick, but -
1. Only 1 wheel is built now
2. SDist is built and uploaded too
```suggestion
          name: distributions
```

---

Yeah it was a wheel before the previous requested changes

---

Actually I checked, and that command does make a wheel

---

I meant that it builds both wheel + SDist. PyBaMM used to build the wheel and the SDist as two different steps before the C++ part was removed (IIRC). Both the distributions (wheel and SDist) are uploaded to PyPI and users can choose which one they want to use (wheel is the default one). It would be wrong to call it "wheels" given that the artifact has a source distribution too (I hope I am not doing something wrong, but that is how it should be).

---

Don't think the name matters but I will change it